float64[] xyz
---
float64[] joint_angles